Created automated test TC_ADMIN_1_9:

j-ororke opened the pull request:
Follows test steps in test plan PR #[4267](https://github.com/CHIP-Specifications/chip-test-plans/pull/4267):
- TH1 able to open commissioning window on DUT
- TH2 Fails to connect to endpoint for 20 iterations with correct payload but incorrect passcode 
- TH2 Fails to connect to endpoint due to commissioning window being closed after the 20th failed attempt 
- TH1 able to open commissioning window on the endpoint again 
- TH1 able to revoke commissioning window on the endpoint

j-ororke committed:
Created automated test TC_ADMIN_1_9:
- Follows test steps in test plan PR #https://github.com/CHIP-Specifications/chip-test-plans/pull/4267:
-- TH1 able to open commissioning window on DUT
-- TH2 Fails to connect to endpoint for 20 iterations with correct payload but incorrect passcode
-- TH2 Fails to connect to endpoint due to commissioning window being closed after the 20th failed attempt
-- TH1 able to open commissioning window on the endpoint again
-- TH1 able to revoke commissioning window on the endpoint

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updated TC_CADMIN_1_9 test:
- Replaced method to get incorrect pincode with using correct pincode in test step 4.

cecille commented on src/python_testing/TC_CADMIN_1_9.py:
```diff
@@ -0,0 +1,113 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
```
```suggestion
#    Copyright (c) 2024 Project CHIP Authors
```

cecille commented on src/python_testing/TC_CADMIN_1_9.py:
```diff
@@ -0,0 +1,113 @@
+                asserts.assert_false(errcode.is_success, 'Commissioning complete did not error as expected')
+                asserts.assert_true(errcode.sdk_code == expectedErrCode,
+                                    'Unexpected error code returned from CommissioningComplete')
+                sleep(1)
```
Do you actually need this sleep for the test to work? If so, can drop the error logs into the slack channel?

j-ororke replied:
Hi Cecille,
So, I had found during testing that if I did not add the sleep time duration after the 20th iteration that the 21st iteration appeared to fail similarly to how it does for iterations 1-20, then it would timeout on the 22nd iteration. So, my thought was adding the sleep duration of 1 second allows the endpoint enough time to close the commissioning window before another attempt can be made to it. 
This also means that the 1 second sleep time does not need to be made for every iteration, so I will move this sleep time out of the for loop where it currently is.
Please let me know if this is acceptable.

j-ororke replied:
After discussing this with you, I will create a bug for the issue found above and place a comment in the test code for why the sleep is there.
Will also move the sleep into the main section of the test code and out of this function.

j-ororke replied:
Created issue #34383 regarding failure noticed when using correct passcode during test step 4.

cecille commented on src/python_testing/TC_CADMIN_1_9.py:
```diff
@@ -0,0 +1,113 @@
+            asserts.assert_false(errcode.is_success, 'Commissioning complete did not error as expected')
+            asserts.assert_true(errcode.sdk_code == expectedErrCode, 'Unexpected error code returned from CommissioningComplete')
+
+    def TC_CADMIN_1_9(self) -> list[str]:
```
```suggestion
    def pics_TC_CADMIN_1_9(self) -> list[str]:
```

cecille commented on src/python_testing/TC_CADMIN_1_9.py:
```diff
@@ -0,0 +1,113 @@
+        return ["CADMIN.S"]
+
+    @async_test_body
+    async def test_TC_ADMIN_1_9(self):
```
```suggestion
    async def test_TC_CADMIN_1_9(self):
```

cecille commented on src/python_testing/TC_CADMIN_1_9.py:
```diff
@@ -0,0 +1,113 @@
+
+    @async_test_body
+    async def test_TC_ADMIN_1_9(self):
+        self.print_step(1, "Commissioning, already done")
```
Could you please move these into a separate steps_ method? That will let the TH follow along as it goes. See https://project-chip.github.io/connectedhomeip-doc/testing/python.html#test-harness-helpers for an explanation, https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TC_TMP_2_1.py for a good example.

j-ororke replied:
Hi Cecille,
I have updated the method for the test steps following the example you provided in TC_TMP_2_1.

cecille commented on src/python_testing/TC_CADMIN_1_9.py:
```diff
@@ -0,0 +1,113 @@
+    def OpenCommissioningWindow(self) -> CommissioningParameters:
+        try:
+            params = self.th1.OpenCommissioningWindow(
+                nodeid=self.dut_node_id, timeout=900, iteration=10000, discriminator=self.discriminator, option=1)
```
oop, sorry, just one more thing - I added a step to the test plan just before it went in to change this from 900 to the max failsafe value. TP is merged now, see step 2. https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/multiplefabrics.adoc#tc-cadmin-1-9-device-exit-commissioning-mode-after-20-failed-commission-attempts-ecm-dut-commissionee

j-ororke replied:
Hi Cecille,
I have updated the duration for the commissioning window here now mentioned in the test plan for test step 2.

cecille commented on src/python_testing/TC_CADMIN_1_9.py:
```diff
@@ -0,0 +1,113 @@
+#
```
You'll also need to add this to tests.yaml (assuming you get this in before Vatsal implements file globbing)

j-ororke replied:
Hi Cecille,
I have updated tests.yaml and added this test into the repl_tests_linux section.

j-ororke committed:
Update src/python_testing/TC_CADMIN_1_9.py

Changing test async function to correct name, thank you Cecille!

Co-authored-by: C Freeman <cecille@google.com>

j-ororke committed:
Update src/python_testing/TC_CADMIN_1_9.py

Co-authored-by: C Freeman <cecille@google.com>

j-ororke committed:
Update src/python_testing/TC_CADMIN_1_9.py

Co-authored-by: C Freeman <cecille@google.com>

j-ororke committed:
Merge branch 'master' into commissioning_failover_check

j-ororke committed:
Updated tests.yaml and TC_CADMIN_1_9.py:
- Added test module TC_CADMIN_1_9 into tests.yaml script
- Created new steps method to contain test steps, then call those in variables in main test code area.
- Inserted comment for why sleep time duration is needed between test steps 3 and 4, moved sleep time duration into main test code area
- Added setting variable for duration of commissioning window using maxCumulativeFailsafeSeconds

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updated TC_CADMIN_1_9 test module:
- added CI argument blocks to attempt to get test to run in CI

j-ororke committed:
Updated TC_CADMIN_1_9:
- Changed method for setting setupPinCode as in CI/CD pipeline it was failing previously.

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into commissioning_failover_check

j-ororke committed:
Updated TC_CADMIN_1_9:
- Changed factory reset to false to attempt running in CI/CD pipeline

j-ororke committed:
Updated TC_CADMIN_1_9:
- Changed factory reset back to true as commissioning failed in CI/CD pipeline without it being set to true

j-ororke committed:
Updated TC_CADMIN_1_9:
- Removed PICS call in CI/CD as not needed
- Turned off quiet mode when running test in CI/CD pipeline to help flush out current issue

j-ororke committed:
Updated TC_CADMIN_1_9:
- Added back removing PICS from script args for running CI/CD pipeline to see if this gets pipeline working again.

j-ororke committed:
Merge branch 'master' into commissioning_failover_check

j-ororke committed:
Updated TC_CADMIN_1_9:
- Added an additional await to see if we can get test to pass in CI/CD pipeline

j-ororke committed:
Updated TC_CADMIN_1_9:
- Changed method in which CommissionOnNetwork was being established, needed to run through asserts module.
- Added await async method for opening commissioning window.

j-ororke committed:
Updated TC_CADMIN_1_9:
- Turned quiet mode back on for CI/CD pipeline for this test as test is now passing.

j-ororke committed:
Merge branch 'master' into commissioning_failover_check

j-ororke committed:
Merge branch 'master' into commissioning_failover_check

j-ororke committed:
Merge branch 'master' into commissioning_failover_check

KishokG commented:
@j-ororke Could you please put the PR to remove TC-CADMIN-1.9 YAML script. Here is the issue for the same: https://github.com/project-chip/matter-test-scripts/issues/363
